sktime/forecasting/model_selection/_split.py

```diff
@@ -412,41 +414,234 @@ def __init__(
         self.fh = fh
 
     def split(self, y: ACCEPTED_Y_TYPES) -> SPLIT_GENERATOR_TYPE:
-        """Split `y` into training and test windows.
+        """Get iloc references to train/test slits of `y`.
 
         Parameters
         ----------
-        y : pd.Series or pd.Index
-            Time series to split
+        y : pd.Index or time series in sktime compatible time series format (any)
+            Index of time series to split, or time series to split
+            If time series, considered as index of equivalent pandas type container:
+                pd.DataFrame, pd.Series, pd-multiindex, or pd_multiindex_hier mtype
 
         Yields
         ------
-        train : np.ndarray
-            Training window indices
-        test : np.ndarray
-            Test window indices
+        train : 1D np.ndarray of dtype int
+            Training window indices, iloc references to training indices in y
+        test : 1D np.ndarray of dtype int
+            Test window indices, iloc references to test indices in y
         """
-        y_index = get_index_for_series(y)
-        for train, test in self._split(y_index):
+        y_index = self._coerce_to_index(y)
+
+        if not isinstance(y_index, pd.MultiIndex):
+            split = self._split
+        else:
+            split = self._split_vectorized
+
+        for train, test in split(y_index):
             yield train[train >= 0], test[test >= 0]
 
     def _split(self, y: pd.Index) -> SPLIT_GENERATOR_TYPE:
-        """Split `y` into training and test windows.
+        """Get iloc references to train/test splits of `y`.
+
+        private _split containing the core logic, called from split
 
         Parameters
         ----------
-        y : pd.Index
-            Index of time series to split
+        y : pd.Index or time series in sktime compatible time series format
+            Time series to split, or index of time series to split
 
         Yields
         ------
-        training_window : np.ndarray
-            Training window indices
-        test_window : np.ndarray
-            Test window indices
+        train : 1D np.ndarray of dtype int
+            Training window indices, iloc references to training indices in y
+        test : 1D np.ndarray of dtype int
+            Test window indices, iloc references to test indices in y
         """
         raise NotImplementedError("abstract method")
 
+    def _split_vectorized(self, y: pd.Index) -> SPLIT_GENERATOR_TYPE:
```

Comment:
Is `pd.MultiIndex` so different from `pd.Index` that this function hast to be separate from `_split`? Or, in other words, is `_split_vectorized` not general enough to cover a simple `pd.Index` too? Or, yet in other words, what would happen if you completely replace `_split` with `_split_vectorized` without differentiating between `pd.MultiIndex` and `pd.Index`?

Reply:
>Or, yet in other words, what would happen 

It would break, because type and methods-wise, `pd.MultiIndex` is different from ordinary `pd.Index` subtypes?
Further, I think it is safer use `_split` directly when it can be used.

Reply:
ok, understood. thanks for clarification!